sklearn/decomposition/_pca.py

```diff
@@ -93,6 +93,7 @@ def _assess_dimension(spectrum, rank, n_samples):
                 (spectrum[i] - spectrum[j]) * (1.0 / spectrum_[j] - 1.0 / spectrum_[i])
             ) + log(n_samples)
 
+    pu = xp.asarray(pu, device=device(spectrum), dtype=spectrum.dtype)
```

Comment:
Is this line still required? Can't we just use a regular Python scalar here? No big deal though.

Reply:
Yes and no. We need it, unless we change from using `scipy.special.gammaln` to `math.lgamma` on L74. With the scipy `gammaln` we get an array with a single element. And if that array is on a "GPU device" then we can't implicitly convert it to a numpy array (which is what `pu` is).

Overall I think it makes sense to just use `math.lgamma`, as we don't want to compute on arrays and even the scipy docs say that they are the same.

Reply:
I think either option should be fine.